lib/puppet/provider/exec/powershell.rb

```diff
@@ -68,7 +68,8 @@ def run(command, check = false)
         return super("cmd.exe /c \"\"#{native_path(command(:powershell))}\" #{legacy_args} -Command - < \"#{native_path}\"\"", check)
       end
     else
-      result = ps_manager.execute(command)
+      timeout_ms = resource[:timeout].nil? ? nil : resource[:timeout].to_i * 1000
```

Comment:
That being said....if https://github.com/puppetlabs/puppetlabs-powershell/pull/125 is merged prior, it does the integer conversion for us so the `to_i` is redundant.